### PR TITLE
Video thumbnail 

### PR DESCRIPTION
**Pull request by davivel**
Replaces #1035, after rebase on master, and targeting master instead of good-old develop.

### CODE REVIEW

File: src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java [hunk 1]
```diff
@@ -391,6 +406,51 @@ public static ThumbnailGenerationTask getBitmapWorkerTask(ImageView imageView) {
         return null;
     }
 
+    public static Bitmap addVideoOverlay(Bitmap thumbnail){
+        Bitmap playButton = BitmapFactory.decodeResource(MainApp.getAppContext().getResources(),
```
Comment (davivel) [inline src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java]: This shouldn't be done in main thread

File: src/com/owncloud/android/ui/adapter/FileListListAdapter.java [hunk 2]
```diff
@@ -314,14 +323,18 @@ public View getView(int position, View convertView, ViewGroup parent) {
                                             fileIcon, mStorageManager, mAccount
                                     );
                             if (thumbnail == null) {
-                                thumbnail = ThumbnailsCacheManager.mDefaultImg;
+                                Integer id = MimetypeIconUtil.getFileTypeIconId(
+                                    file.getMimetype(),
+                                    file.getFileName()
+                                );
+                                thumbnail = BitmapFactory.decodeResource(mContext.getResources(), id);
```
Comment (davivel) [inline src/com/owncloud/android/ui/adapter/FileListListAdapter.java]: This shouldn't be done in main thread

File: src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java [hunk 3]
```diff
@@ -180,8 +183,20 @@ protected Bitmap doInBackground(Object... params) {
...
+
+                    if (((OCFile) mFile).isVideo() && thumbnail != null){
+                        thumbnail = addVideoOverlay(thumbnail);
+                    }
                 }  else if (mFile instanceof File) {
                     thumbnail = doFileInBackground();
+
+                    String url = ((File) mFile).getAbsolutePath();
+                    FileNameMap fileNameMap = URLConnection.getFileNameMap();
+                    String mMimeType = fileNameMap.getContentTypeFor("file://" + url);
```
Comment (davivel) [inline src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java]: To be replaced with MimeTypeIconUtil.getBestMimeTypeByFilename(...) instead.

**Comment by davivel** [comment]
@jesmrec , reviewed, some changes needed.

**Comment by jesmrec** [comment]
Pending server side support

**Comment by hannesa2** [comment]
> Pending server side support

Still pending ? @jesmrec

**Comment by hannesa2** [comment]
I get no thumbnail back (error 404) and [here](https://github.com/owncloud/android-library/pull/286/files#r449374792) the switch was removed. Does is no more works at all on server side and all is senseless ?

Or what I've to do, to make it work to avoid a 404 ?
https://www.myserver.info/owncloud/index.php/apps/files/api/v1/thumbnail/448/448/CameraUpload/VID_20200418_164100.mp4

**Comment by jesmrec** [comment]
This PR was blocked time ago because more elaboration was needed on the server side. We keep it, just in case any day and will be posible to move forward.

**Comment by hannesa2** [comment]
> This PR was blocked time ago because more elaboration was needed on the server side. We keep it, just in case any day and will be posible to move forward.

Is there a github issue on server side to link with this ?

**Comment by michaelstingl** [comment]
> Is there a github issue on server side to link with this ?

I think this is the most relevant one: https://github.com/owncloud/core/issues/24480

Foundational stuff has been done here: https://github.com/owncloud/core/pull/24975

I was using this search term to find the traces:
https://github.com/issues?q=org%3Aowncloud+ffmpeg

**Comment by abelgardep** [comment]
Thumbnails will be retrieved from server via preview api.

**Pull request closed without merging**